theia/adapters/usgs/eros_wrapper.py

```diff
@@ -62,40 +62,52 @@ def parse_result_set(cls, result_set):
         return [scene.get('displayId', None) for scene in result_set if 'displayId' in scene]
 
     @classmethod
-    def eros_get(cls, url, request_data, **kwargs):
+    def eros_get(cls, url, request_data, authenticating=False, **kwargs):
         if url != 'login' and (not cls.token()):
             cls.connect()
 
-        new_args = cls.eros_prepare(request_data, **kwargs)
+        new_args = cls.eros_prepare(request_data, authenticating=authenticating, **kwargs)
         return requests.get(cls.api_url(url), **new_args).json()
 
     @classmethod
-    def eros_post(cls, url, request_data, **kwargs):
+    def eros_post(cls, url, request_data, authenticating=False, **kwargs):
         if url != 'login' and (not cls.token()):
             cls.connect()
 
-        new_args = cls.eros_prepare(request_data, **kwargs)
+        new_args = cls.eros_prepare(request_data, authenticating=authenticating, **kwargs)
         return requests.post(cls.api_url(url), **new_args).json()
 
     @classmethod
-    def eros_prepare(cls, request_data, **kwargs):
-        headers = {'Content-Type': 'application/json'}
+    def eros_prepare(cls, request_data, authenticating=False, **kwargs, ):
+        headers = {}
         if cls.token():
             headers['X-Auth-Token'] = cls.token()
         if 'headers' in kwargs:
             headers = {**headers, **(kwargs['headers'])}
 
         params = {}
-
-        if request_data:
-            params = {'jsonRequest': json.dumps(request_data)}
-
         if 'params' in kwargs:
             params = {**params, **(kwargs['params'])}
 
-        new_args = {
-            'headers': headers,
-            'params': params,
-        }
+        if authenticating:
+            headers['Content-Type'] = 'application/x-www-form-urlencoded'
+
+            if request_data:
+                data = 'jsonRequest=' + quote(json.dumps(request_data))
+
+            new_args = {
+                'headers': headers,
+                'data': data,
```

Comment:
Doesn't look like this is instantiated anywhere if no request data is provided like `params` is up there.

Reply:
Yeah, a few of these things I'm not quite sure about 🤔still trying to figure out exactly how this app works and which things need to be there for "django magic" to occur.